### DISCUSSION TITLE
ImportError: cannot import name 'DataError' from 'pandas.core.base'

### DISCUSSION THREAD

**Discussion by etanaga** [Data-Centric-AI-Community/fg-data-profiling#1066] [Q&A]
Hi all.
I have an issue with pandas_profiling, currently my environment  

- Conda 4.14.0 
- Python 3.9.13
- pandas 1.5.0
- pandas-profiling 2.9.0

And received an error while try to import pandas_profilling

from pandas_profiling import ProfileReport
profile = ProfileReport(retail_raw)
profile.to_notebook_iframe()

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [11], in <cell line: 1>()
----> 1 from pandas_profiling import ProfileReport
      2 profile = ProfileReport(retail_raw)
      3 profile.to_notebook_iframe()

File ~\Anaconda3\lib\site-packages\pandas_profiling\__init__.py:7, in <module>
      1 """Main module of pandas-profiling.
      2 
      3 .. include:: ../../README.md
      4 """
      6 from pandas_profiling.config import Config, config
----> 7 from pandas_profiling.controller import pandas_decorator
      8 from pandas_profiling.profile_report import ProfileReport
      9 from pandas_profiling.version import __version__

File ~\Anaconda3\lib\site-packages\pandas_profiling\controller\pandas_decorator.py:4, in <module>
      1 """This file add the decorator on the DataFrame object."""
      2 from pandas import DataFrame
----> 4 from pandas_profiling.__init__ import ProfileReport
      7 def profile_report(df, **kwargs) -> ProfileReport:
      8     """Profile a DataFrame.
      9 
     10     Args:
   (...)
     15         A ProfileReport of the DataFrame.
     16     """

File ~\Anaconda3\lib\site-packages\pandas_profiling\__init__.py:8, in <module>
      6 from pandas_profiling.config import Config, config
      7 from pandas_profiling.controller import pandas_decorator
----> 8 from pandas_profiling.profile_report import ProfileReport
      9 from pandas_profiling.version import __version__
     11 clear_config = ProfileReport.clear_config

File ~\Anaconda3\lib\site-packages\pandas_profiling\profile_report.py:11, in <module>
      8 from tqdm.auto import tqdm
     10 from pandas_profiling.config import config
---> 11 from pandas_profiling.model.describe import describe as describe_df
     12 from pandas_profiling.model.messages import MessageType
     13 from pandas_profiling.report import get_report_structure

File ~\Anaconda3\lib\site-packages\pandas_profiling\model\describe.py:11, in <module>
      9 from pandas_profiling.config import config as config
     10 from pandas_profiling.model.base import Variable
---> 11 from pandas_profiling.model.correlations import calculate_correlation
     12 from pandas_profiling.model.sample import Sample, get_sample
     13 from pandas_profiling.model.summary import (
     14     get_duplicates,
     15     get_messages,
   (...)
     19     get_table_stats,
     20 )

File ~\Anaconda3\lib\site-packages\pandas_profiling\model\correlations.py:11, in <module>
      9 import pandas as pd
     10 from confuse import NotFoundError
---> 11 from pandas.core.base import DataError
     12 from scipy import stats
     14 from pandas_profiling.config import config

ImportError: cannot import name 'DataError' from 'pandas.core.base' (C:\Users\etana\Anaconda3\lib\site-packages\pandas\core\base.py)

Need the help to fix my problem.
Thanks

**Comment by corytomlinson** [comment]
Same issue here. Python 3.10.

**Comment by floriancircly** [comment]
I had the same issue with pandas-profiling="3.2.0". I updated the module to pandas-profiling="3.3.0" and it worked without any further issues. (pandas="1.4.4" and Python 3.8.10)

**Comment by fabclmnt** [comment]
Yes this is already sorted in the latest version of pandas-profiling 3.2.0 and 3.4.0. 

I'll ask you to confirm me whether with the newest versions and in a conda or pyenv, the problem is sorted.

**Comment by learnnk** [comment]
I'm facing the same issue with below versions. How can i resolve the error any help much appreciated
pandas                1.5.3
pandas-profiling      3.1.0
ImportError: cannot import name 'DataError' from 'pandas.core.base'
(C:\Users\Username\PycharmProjects\DataQuality\venv\lib\site-packages\pandas\core\base.py

**Reply by fabclmnt** [reply]
Hi @learnnk,

this is an issue solved in more recent versions. Can you please test with versions above 3.2.0 ?

**Comment by muha99** [comment]
hey guys i have installed pandas_profiling and when i run my code it says no module named pandas_profiling python 3.12.0

**Reply by fabclmnt** [reply]
Hi @muha99 

you should be using ydata-profiling instead, as per the current documentation. 

ydata-profiling is not yet available for python 3.12 only until python 3.11.

**Comment by muha99** [comment]
thanks for your reply , i have tried to install that library but it stars the installation but at the end it gives me this  Building wheel for levenshtein (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for levenshtein (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [27 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\Levenshtein
      copying Levenshtein\StringMatcher.py -> build\lib.win-amd64-cpython-311\Levenshtein
      copying Levenshtein\__init__.py -> build\lib.win-amd64-cpython-311\Levenshtein
      running egg_info
      writing levenshtein.egg-info\PKG-INFO
      writing dependency_links to levenshtein.egg-info\dependency_links.txt
      writing namespace_packages to levenshtein.egg-info\namespace_packages.txt
      writing requirements to levenshtein.egg-info\requires.txt
      writing top-level names to levenshtein.egg-info\top_level.txt
      reading manifest file 'levenshtein.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*pyc' found anywhere in distribution
      warning: no previously-included files matching '*so' found anywhere in distribution
      warning: no previously-included files matching '.project' found anywhere in distribution
      warning: no previously-included files matching '.pydevproject' found anywhere in distribution
      adding license file 'COPYING'
      writing manifest file 'levenshtein.egg-info\SOURCES.txt'
      copying Levenshtein\_levenshtein.c -> build\lib.win-amd64-cpython-311\Levenshtein
      copying Levenshtein\_levenshtein.h -> build\lib.win-amd64-cpython-311\Levenshtein
      running build_ext
      building 'Levenshtein._levenshtein' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for levenshtein 
and when i run the code using ydataprep it says ModuleNotFoundError: No module named 'dataprep'

**Comment by muha99** [comment]
i have tried a lot of solutions , i have created a venv for both python 3.11 and 3.12 but it doesn't work i have installed pandas_profiling for python 3.12 it was installed but after running my code it says  No module named 'pandas.... i have upgraded pip , pandas and everything , i also tried deactivating my antivirus because sometimes it doesn't let some libraries to be installed but nothing works for me
